src/Approximations/decompositions.jl

```diff
@@ -442,3 +478,29 @@ The template direction approximation of the projection of `S`.
     M = sparse(1:length(block), block, ones(N, length(block)), length(block), n)
     return overapproximate(M * S, directions)
 end
+
+"""
+    project(S::LazySet{N},
+            block::AbstractVector{Int},
+            n::Int
+           )::HPolytope where {N<:Real}
+
+Project a high-dimensional set to a low-dimensional set.
+
+### Input
+
+- `S` -- set
+- `block` -- block structure - a vector with the dimensions of interest
+- `n` -- (optional, default: `dim(S)`) ambient dimension of the set `S`
+
+### Output
+
+The lazy linear map of the projection of `S`.
+"""
```

Comment:
it doesn't hurt to add an example here.

it also seems that the input parameter `n` can be removed: if `n` is different  from `dim(S)`, then `M * S` will break anyways:

```julia
julia> rand(2, 2) * rand(Interval)
ERROR: AssertionError: a linear map of size (2, 2) cannot be applied to a set of dimension 1
Stacktrace:
 [1] LinearMap{Float64,Interval{Float64,IntervalArithmetic.Interval{Float64}},Float64,Array{Float64,2}}(::Array{Float64,2}, ::Interval{Float64,IntervalArithmetic.Interval{Float64}}) at /Users/forets/.julia/dev/LazySets/src/LinearMap.jl:32
 [2] Type at /Users/forets/.julia/dev/LazySets/src/LinearMap.jl:40 [inlined]
 [3] *(::Array{Float64,2}, ::Interval{Float64,IntervalArithmetic.Interval{Float64}}) at /Users/forets/.julia/dev/LazySets/src/LinearMap.jl:65
 [4] top-level scope at none:0
```

Reply:
We added `n` to all `project` methods because the caller might already know `n`. In general, can has to be computed (e.g., for a `CartesianProductArray`).

Reply:
Ok. I forgot that we did this as a workaround for caching the dimension of the set. The problem is that passing anything different from `dim(S)` will produce an error. In general computing `dim(..)` shouldn't take more than a few microseconds, unless the set is very nested.

Reply:
My point is: That discussion is completely independent from this PR. If you want to remove `n`, we should remove it everywhere.